### PR TITLE
Allow previewing actual weapon damage

### PR DESCRIPTION
**Pull request by Coolthulhu**
Functions like `item::melee_value` are bad estimates of actual usability in melee, which makes NPC weapon selection quite bad (see #13315 ).

Eventually I'll need to rewrite them to actually use damage calculations. Those calculations could be shown to the player somewhere, to make all the arcane mechanics less arcane and make effects of balancing quite obvious.

A typical output at the moment looks like this:

```
Critical chance vs slow enemy: 75%
Critical chance vs fast enemy: 40%
10 bashing, 20 on crit
3 piercing, 3 on crit
```

With the slow/fast being about dodging (you get more crits if your `hit_roll > enemy_dodge_roll * 1.5`).

Now, would it be OK to display all the stats to players?
If not, should it be a debug-only option or should it be partially revealed through arcane comments like "you occasionally/often/always get critical hits with this item"?

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
We already see approx (or average?) damage output of weapons, why not show us full version?
Also, full potential of weapon can be discovered after some practice and leveling up in combat skills.

**Comment by narc0tiq** [comment]
I like the progressive disclosure idea (reveal more info as you use the
item more/gain higher skill). If we end up displaying this data to players,
probably apply some fuzzing to the shown values, based on skill (lower
skill == more fuzzing).

Mostly, I'm of the opinion that I like visible mechanics more than I like
invisible ones (encumbrance still does my head in, and that's fairly
well-discussed at this point), so I'd like to see the numbers, but at the
same time doing that breaks down the fourth wall somewhat. Descriptors like
"You expect to often get critical hits with this item." work better from
that point of view. A compromise can be including both, with the numeric
value in parentheses.

**Comment by Coolthulhu** [comment]
Went with skill in melee >= 3. For now no fuzzing, can be added later since all.

**Comment by narc0tiq** [comment]
I rather think I like it, but would like more opinions, as well.

**Comment by Zireael07** [comment]
Sweet!

**Comment by Rivet-the-Zombie** [comment]
I like it!

**Comment by Rivet-the-Zombie** [comment]
I like it!

**Comment by stk2008** [comment]
I like it to

Mainly because it allows npcs to choose a better weapon?.

But yes I also like being able to see detailed info

**Comment by narc0tiq** [comment]
NPCs don't read, silly.

**Comment by stk2008** [comment]
I know they can't read but I assumed from the part that says

Makes npc weapon selection bad I thought it had some kind of way to pass
info to npcs.

Any way hey narc  long time no speak :)

**Comment by Coolthulhu** [comment]
It doesn't fix the NPCs yet, it's the first part. Later I'll add it to NPC weapon evaluation, but I'll have to figure out a way to make it execute faster than it does now now.

**Comment by narc0tiq** [comment]
Rawr, I know, it just seemed like a funny comment. I'm not sure I pulled it
off, though.

Blame sleep deprivation.

And yes, since there is now a way for the code to estimate a weapon's
damage better, it lets NPCs pick weapons that are actually _good_ from all
the stuff they own.

**Comment by stk2008** [comment]
Ah I see.

Anyways love your work mate the z level stuff your doing is amazeing

**Comment by narc0tiq** [comment]
Heh, okay, correction: it *could* let NPCs pick actually good weapons. Eventually.

**Comment by stk2008** [comment]
U know me narc  :p I'm good at saying silly things :).

Like Jenkins is always on irc later to find out its a bot :(.

Oh well love the cdda community.

**Comment by stk2008** [comment]
Well that's what I thought that's how I read it any way so I kind of was
right :p.

File: src/item.cpp [hunk 1]
```diff
@@ -1583,8 +1615,10 @@ std::string item::info(bool showtext, std::vector<iteminfo> &dump_ref) const
                 }
             }
             if (known_recipes.size() > 24) {
+                dump->push_back(iteminfo("DESCRIPTION", "--"));
```
Comment (kevingranade) [inline src/item.cpp]: Is this intentional? I seem to remember it being removed recently.

Comment (Coolthulhu) [inline src/item.cpp]: I could add it above (in the damage preview) or have two sections next to each other.
I don't really see the reason for omitting the newline/"--" before this section. I'd say it looks better with it.

**Pull request merged**